csharp/src/Ice/Connection.cs

```diff
@@ -1156,14 +1156,14 @@ private async Task CloseAsync(Exception? exception)
                     case Ice1Definitions.FrameType.ValidateConnection:
                     {
                         ProtocolTrace.TraceReceived(_communicator, Endpoint.Protocol, readBuffer);
-                        if (_heartbeatCallback != null)
+                        var hearbetReceived = _heartbeatReceived;
```

Comment:
Should be `heartbeatReceived`. Why are you making this copy?

Reply:
We can avoid the copy and just check for null on Invoke

```
incoming = () =>
{
    try
    {
        _heartbeatReceived?.Invoke(this, EventArgs.Empty);
```